src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -904,6 +904,31 @@ begin
   ... ≤ R + δ : add_le_add (hR zE) hz.le
 end
 
+/-- The frontier of the (open) thickening of a set is contained in an `inf_edist` level set. -/
+lemma frontier_thickening_subset (E : set α) {δ : ℝ} :
+  frontier (thickening δ E) ⊆ {x : α | inf_edist x E = ennreal.of_real δ} :=
+frontier_lt_subset_eq continuous_inf_edist continuous_const
+
+lemma frontier_thickening_disjoint (A : set X) :
+  pairwise (disjoint on (λ (r : ℝ), frontier (thickening r A))) :=
+begin
+  refine (pairwise_disjoint_on _).2 (λ r₁ r₂ hr, _),
+  cases le_total r₁ 0 with h₁ h₁,
+  { simp [thickening_of_nonpos h₁] },
+  refine ((disjoint_singleton.2 $ λ h, hr.ne _).preimage _).mono
+    (frontier_thickening_subset _) (frontier_thickening_subset _),
+  apply_fun ennreal.to_real at h,
+  rwa [ennreal.to_real_of_real h₁, ennreal.to_real_of_real (h₁.trans hr.le)] at h
+end
+
+lemma frontier_thickening_disjoint' (A : set X) (radii : set ℝ) :
+  pairwise (disjoint on (λ (r : radii), frontier (thickening r A))) :=
```

Comment:
Please use the `set.pairwise_disjoint` predicate.

Reply:
Thanks, this predicate seems appropriate. But the proof of the special case becomes a one-liner, so I will rethink whether it should just be done so within a later proof.

I will remove this version for now from the PR.